substrate/frame/staking/ahm-test/src/lib.rs

```diff
@@ -0,0 +1,127 @@
+#[cfg(test)]
+pub mod ah;
+#[cfg(test)]
+pub mod rc;
+
+#[cfg(test)]
+pub mod shared;
+
+// shared tests.
+#[cfg(test)]
+mod tests {
+	use super::*;
+	use codec::Decode;
+	use frame::testing_prelude::*;
+	use pallet_election_provider_multi_block as multi_block;
+	use pallet_staking::{ActiveEra, ActiveEraInfo};
+
+	#[test]
+	fn rc_session_change_reported_to_ah() {
```

Comment:
@Ank4n @tdimitrov if you run this test with `RUST_LOG=runtime=debug` you will see somewhat of a working e2e example.